discovery-provider/src/tasks/playlists.py

```diff
@@ -208,13 +208,19 @@ def parse_playlist_event(
 
             playlist_content_array = []
             for track_id in event_args._orderedTrackIds:
-                track_time_array_length = len(intermediate_track_time_lookup_dict[track_id])
-                if track_time_array_length > 1:
-                    track_time = intermediate_track_time_lookup_dict[track_id].pop(0)
-                elif track_time_array_length == 1:
-                    track_time = intermediate_track_time_lookup_dict[track_id][0]
-                else:
+                if track_id not in intermediate_track_time_lookup_dict:
```

Comment:
nit, i'd switch the order of cases here -- 

if track_id in ... : 
   do the stuff
else:
  logger.info(not found)
  track_time = block_integer_time

makes for a little easier reasoning

Reply:
agreed, will make this change